django/utils/metaclassmaker.py

```diff
@@ -0,0 +1,87 @@
+"""
+ Complex metaclass merger library for solving multiple metadatas inheritance.
+ Main code and idea is from:
+ http://code.activestate.com/recipes/204197-solving-the-metaclass-conflict/
+"""
+from __future__ import absolute_import
+
+import inspect
+
+from django.utils import six
+
+memoized_metaclasses_map = {}
+
+
+def _skip_redundant(iterable, skip_set=None):
+    """Redundant items are repeated items or items in the original skip_set."""
```

Comment:
suggested wording: "Yield each item in iterable at most once and exclude those that appear in skip_set."